File: Dataset/Leetcode/test/3/542.cpp
 class Solution {
public:
    int XXX(string s) {
        int res,count = 0;
        string str;//当前子字符串
        for(auto c:s){
            if(str.find(c)==string::npos){
                str += c;
                count++;
                continue;
            }
            else {
                res = count>res? count:res;
                count = 0;
                str = "";
                str += c;
            }
        }
        return res;
    }
};

File: Dataset/Leetcode/valid/22/542.py
 class Solution:
    def XXX(self, n: int) -> List[str]:
        res = []
        temp = []
        def go(temp, i, step):
            if len(temp) == 2*n and i == 0:
                s = ''.join(temp)
                if s not in res:
                    res.append(s)
            if step <= 2*n:
                if i == 0:
                    go(temp+['('],i+1,step+1)
                elif i < n:
                    go(temp+['('],i+1,step+1)
                    go(temp+[')'],i-1,step+1)
                elif i == n:
                    go(temp+[')'],i-1,step+1)
        go(temp,0,1)
        return res

